src/machines/triviaMachine.ts

```diff
@@ -0,0 +1,252 @@
+import { assign, fromPromise, not, setup } from "xstate";
+import { RMCharacter } from "../common/types";
+import { RickCharacters } from "../services/RickApi";
+import { getRandomNumber } from "../common/constants";;
+
+const characterLogic = fromPromise(async () => {
+    const randomNumber = getRandomNumber();
+    const character = await RickCharacters.getCharacter(randomNumber)
+    return character;
+});
+
+
+const triviaMachine = setup({
+    types: {
+        events: {} as 
+        | { type: "user.play"} 
+        | { type: "user.close"} 
+        | { type: "user.reject"}
+        | { type: "user.accept"}
+        | { 
+            type: "user.selectAnswer",
+            answer: number
+        }
+        | { type: "user.nextQuestion"}
+        | { type: "user.toggleClue"}
+        | { type: "user.playAgain"},
+        context: {} as {
+            homePageCharacters: Array<RMCharacter>
+            hasLoaded: boolean
+            currentCharacter: RMCharacter | null
+            randomCharacters: Array<RMCharacter>
+            isClueOpened: boolean
+            points: number
+            question: number
+            lifes: number
+        }
+    },
+    guards: {
+        isAnswerCorrect: ({context, event}) => {
+            if(event.type !== "user.selectAnswer") return false
+            if(!context.currentCharacter) return false
+            return event.answer === context.currentCharacter.id
+        },
+        hasLostGame: ({context}) => {
+            return context.lifes <= 0
+        },
+        hasWonGame: ({context}) => {
+            return context.points >= 100
+        }
+    },
```

Comment:
Guards are functions that help us define if the app can go from one state to another. They must return a boolean value.